docs/source/transaction_family_specifications/supplychain_transaction_family.rst

```diff
@@ -0,0 +1,317 @@
+        string name = 2; // the agent name
+    }
+
+
+Application
+-----------
+
+An Application is a offer from an agent to change the Custodian or Owner field of a Record.
+
+.. code-block:: protobuf
+
+    message Application {
+
+        string record_identifier = 1; // the natural key of the record
+        string applicant = 2; // public key of the applicant
+        int64 creation_time = 3;
+
+        // The Type indicates the type of this Application
+        enum Type {
+            OWNER = 0;
+            CUSTODIAN = 1;
+        }
+        Type type = 4;
+
+        // The Status indicates the type of this Application
+        enum Status {
+            OPEN = 0;
+            CANCELED = 1;
+            REJECTED = 2;
+            ACCEPTED = 3;
+        }
+        Status status = 5;
+        string terms = 6; // the terms of the application. Human readable.
+    }
+
+
+Record
+------
+A record represents a good or material that is being tracked in the system.
+
+.. code-block:: protobuf
+
+    message Record {
+        string identifier = 1; // the natural key of the record, serial number or
+        // attached sensor identifier
+
+        int64 creation_time = 2; // the time the record was created
+
+        message AgentRecord {
+            string agent_identifier = 1; // the public key of the agent
+            int64 start_time = 2; // the time the agent started in the role
+        }
+        repeated AgentRecord owners = 3; // ordered list of the owners, current
+        // owner is first in the list. List is ordered in cronologic order oldest
+        // to most recent. The first by defition is the creator of the record. The
+        // last is the current owner of the record.
+        repeated AgentRecord custodians = 4; // ordered list of custodians.
+        // current owner is first in the list. Same ordering as the owners list.
+
+        bool final = 5; // is the record finalized, finalized records cannot be
+        // changed.
+
+    }
+
+
+Addressing
+----------
+
+When a setting is read or changed, it is accessed by addressing it using the following algorithm:
+
+The SCTP uses the 3 namespaces, to store the corresonding objects:
+    - supplychain.agent
+    - supplychain.application
+    - supplychain.record
+
+The strings are converted to a Merkle prefix by taking the first 6 characters
+of the hex encoded sha512 hash of the string. .
+
+Agent addresses are generated sha512(agent_pub_key), prepended with the supplychain.agent namespace.
+
+.. code-block:: javascript
+
+    agent_addr = namespace + sha512(agent_pub_key)).hexdigest()
+
+Application addresses are generated as the concatination of the hash(sha256)
+applicant agent public key and the hash(sha256) record_identifier. sha256 is
+used here to so the top half of the address can identify the Agent and the
+bottom half can identify the Record.
+
+.. code-block:: javascript
+
+    agent_addr = namespace + sha256(agent_pub_key)).hexdigest()
```

Comment:
How do you find open applications for a Record? It looks like any Agent can submit applications, so these aren't forward derivable in an efficient way without an index.